tests/test_main_shutdown_coverage.py

```
@@ -0,0 +1,358 @@
+        original_shutting_down = mu.shutting_down
+        original_reconnecting = mu.reconnecting
+        original_reconnect_task = mu.reconnect_task
+        original_reconnect_task_future = mu.reconnect_task_future
+        try:
+            mu.reconnect_task = None
+            mu.reconnect_task_future = None
+
+            original_wait_for = asyncio.wait_for
+            wait_for_call_count = 0
+
+            async def _wait_for_that_times_out_on_gather(coro, timeout):
```

Ruff (0.15.9)

[warning] Missing return type annotation for private function `_wait_for_that_times_out_on_gather`

(ANN202)